examples/path-aliases-v1/renderer/onRenderHtml.tsx

```diff
@@ -1,13 +1,13 @@
 // https://vike.dev/onRenderHtml
-export default onRenderHtml
 
 import ReactDOMServer from 'react-dom/server'
 import React from 'react'
 import { escapeInject, dangerouslySkipEscape } from 'vike/server'
 import { PageLayout } from './PageLayout'
-import type { PageContext } from '#root/types'
+import type { PageContextServer } from '#root/types'
+import type { Config, DocumentHtml } from 'vike/types'
 
-function onRenderHtml(pageContext: PageContext) {
+const onRenderHtml: Config['onRenderHtml'] = (pageContext: PageContextServer): DocumentHtml => {
```

Comment:
@brillout fun fact: I suspect we are benefiting here from the return-type flexibility.

Here I am defining a function of type

```
(pageContext: PageContextServer) => DocumentHtml
```

which ends up satisfying / being assignable to

```
(pageContext: PageContextServer) => DocumentHtml | { documentHtml: DocumentHtml; pageContext?: Partial<Vike.PageContext> | Function }
```